src/NUnitFramework/tests/Constraints/EndsWithConstraintTests.cs

```diff
@@ -24,6 +24,15 @@ public void SetUp()
             new TestCaseData( "say hello to Fred", "\"say hello to Fred\"" ),
             new TestCaseData( string.Empty, "<string.Empty>" ),
             new TestCaseData( null , "null" ) };
+
+        [Test]
+        public void RespectsCulture()
+        {
+            var constraint = new EndsWithConstraint("r\u00E9sum\u00E9");
+
+            var result = constraint.ApplyTo("re\u0301sume\u0301");
```

Comment:
I'm very glad to have found this. It took me a long time to find a single, clear test case which could be applied to demonstrate all the test scenarios in this PR and which worked in both NLS + ICU on all runtimes _locally_, and it looks like the CI agrees too